ykrt/src/mt.rs

```diff
@@ -8,25 +8,24 @@ use std::{
     ptr,
     rc::Rc,
     sync::{
-        atomic::{AtomicBool, AtomicU32, AtomicUsize, Ordering},
+        atomic::{AtomicBool, AtomicUsize, Ordering},
         Arc,
     },
     thread::{self, JoinHandle},
 };
 use yktrace::{start_tracing, ThreadTracer, TracingKind};
 
-pub type HotThreshold = u32;
+pub type HotThreshold = usize;
 const DEFAULT_HOT_THRESHOLD: HotThreshold = 50;
 
 // The current meta-tracing phase of a given location in the end-user's code. Consists of a tag and
-// (optionally) a value. The tags are in the high order bits since we expect the most common tag is
-// PHASE_COMPILED which (one day) will have an index associated with it. By also making that tag
-// 0b00, we allow that index to be accessed without any further operations after the initial
-// tag check.
-const PHASE_TAG: u32 = 0b11 << 30; // All of the other PHASE_ tags must fit in this.
-const PHASE_COMPILED: u32 = 0b00 << 30;
-const PHASE_TRACING: u32 = 0b01 << 30;
-const PHASE_COUNTING: u32 = 0b10 << 30; // The value specifies the current hot count.
+// (optionally) a value. We expect the most commonly encountered tag at run-time is PHASE_COMPILED
+// whose value is a pointer to memory. By also making that tag 0b00, we allow that index to be
+// accessed without any further operations after the initial tag check.
+const PHASE_TAG: usize = 0b11; // All of the other PHASE_ tags must fit in this.
+const PHASE_COMPILED: usize = 0b00;
```

Comment:
Should we put an assertion somewhere that checks that the alignment of our trace code buffers guarantees enough space for our tags?

It almost seems silly because dynasm using `mmap`, which page aligns. Page alignment varies, but is usually 4096, which would give us a whopping 12 bits.

Reply:
> Should we put an assertion somewhere that checks that the alignment of our trace code buffers guarantees enough space for our tags?

I don't think that's possible (how do we actually know the alignment? all we can do is document it and hope) or necessary (in practise, the scheme we're using is tried and tested; or, at least, the scheme in this PR :p ).

Reply:
You can query the page size at runtime and compute the alignment.